SU-HSDO/suhumsci
# SHS-5629: Editors can more easily edit a caption

# READY FOR REVIEW

## Summary
- Adds a patch to Drupal core to add an edit icon next to media items that use the core media library form display.
- This allows for editing of the media item in a modal while in a node edit form.
- Tugboat testing instance: https://pr1580-lqaarerjbqqcvxksaxnggcqcjx0lz4qh.tugboatqa.com

## Need Review By (Date)
7/22/2024

## Urgency
low

## Steps to Test
1. Login to the site 
2. Edit the homepage or a page that has a "Banner image(s) with partial overlay and text

![Screenshot 2024-07-19 at 3 48 55 PM](https://github.com/user-attachments/assets/3c0c0816-ab15-45a5-9d7a-64720921e915)

3. Click "Edit" in the "Banner image(s) with partial overlay and text" paragraph

![Screenshot 2024-07-19 at 3 49 07 PM](https://github.com/user-attachments/assets/ea3db652-456a-4e24-bc19-dc5c67482e7a)

4. Click "Edit" on one of the "Banner image with overlay and text" paragraphs

![Screenshot 2024-07-19 at 3 49 16 PM](https://github.com/user-attachments/assets/c358a293-78f1-4bd3-826c-9a21596d709f)

5. Verify that there's now a pencil icon in addition to the "X" icon on top of the Image media

![Screenshot 2024-07-19 at 3 49 22 PM](https://github.com/user-attachments/assets/a9efe79e-1b02-4a39-8a68-83f0e438881f)

6. Click the pencil icon
7. Verify that a modal pops up that has the media edit form for that image

![Screenshot 2024-07-19 at 3 49 29 PM](https://github.com/user-attachments/assets/0ee87245-a3a4-4bbe-a72d-f337e87bb440)

8. Change the text in the "Caption/Credit" text area
9. Save the media in the modal
10. Save the page
11. Verify that the caption text on the slide that was edited has now changed

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207858777652180
